src/vsphere_cpi/lib/cloud/vsphere/vm_config.rb

```diff
@@ -11,8 +19,28 @@ def upgrade_hw_version?(vmtype_hw_version, global_hw_version)
       vmtype_hw_version.nil? ? global_hw_version : vmtype_hw_version
     end
 
+    # NOTE : This always creates a new name
+    def get_uuid_name
+      @vm_cid = "vm-#{SecureRandom.uuid}"
+      @vm_cid
+    end
+
+    # This returns if the name is already cached.
     def name
-      @vm_cid ||= "vm-#{SecureRandom.uuid}"
+      return @vm_cid if @vm_cid
+      if human_readable_name_enabled? && human_readable_name_info.has_key?('instance_group_name') && human_readable_name_info.has_key?('deployment_name')
```

Comment:
Not sure I like this. We have to do all of these `has_key?` checks because `human_readable_name_info` is a `Hash`. We should probably just make it an actual class or a `Struct` with the `instance_group_name` and `deployment_name` required attributes.